### ISSUE TITLE
[Adjust Rule] SIM111 for compound conditionals

### ISSUE THREAD

**Issue by hjoliver** [MartinThoma/flake8-simplify#59]
## Desired change

* **Rule(s)**: SIM111
* **Adjustment**: Fix warning message for compound conditionals

## Explanation

The warning message is incorrect for compound expressions.  It can be fixed by enclosing the conditional in parentheses.

## Example

This is an example where the mentioned rule(s) would currently be ~~suboptimal~~ wrong:

```python
for x in iterable:
    if A(x) and B(x):
        return False
return True
```
Result, with correction:
```diff
- SIM111: Use 'return all(not A(x) and B(x) for x in iterable)'
+ SIM111: Use 'return all(not (A(x) and B(x)) for x in iterable)'
                              ^             ^
```
or (but you probably don't want to be parsing the expression and doing boolean algebra?):
```diff
- SIM111: Use 'return all(not A(x) and B(x) for x in iterable)'
+ SIM111: Use 'return all(not A(x) or not B(x) for x in iterable)'
```
